symfony/plugins/orangehrmAdminPlugin/client/jobCategory/views/JobCategoryList.vue

```diff
@@ -0,0 +1,175 @@
+        { name: "name", title: "Job Category", style: { "flex-basis": "80%" } },
+        {
+          name: "actions",
+          title: "Actions",
+          style: { "flex-shrink": 1 },
+          cellType: "oxd-table-cell-actions",
+          cellConfig: {
+            delete: {
+              onClick: this.onClickDelete,
+              component: "oxd-icon-button",
+              props: {
+                name: "trash",
+              },
+            },
+            edit: {
+              onClick: this.onClickEdit,
+              props: {
+                name: "pencil-fill",
+              },
+            },
+          },
+        },
+      ],
+      items: [],
+      total: 0,
+      pages: 1,
+      currentPage: 1,
+      pageSize: 5,
+      showPaginator: false,
+      editItem: null,
+      checkedItems: [],
+    };
+  },
+
+  components: {
+    "delete-confirmation": DeleteConfirmationDialog,
+  },
+
+  watch: {
+    currentPage() {
+      this.resetDataTable();
+      this.fetchData();
+    },
+  },
+
+  created() {
+    this.fetchData();
+  },
+
+  methods: {
+    onClickAdd() {
+      this.$emit("onAddItem", { viewId:2, payload: null });
+    },
+    onClickEdit(item) {
+      this.$emit("onEditItem", { viewId:3, payload: item });
+    },
+    onClickDeleteSelected() {
+      const ids = this.checkedItems.map((index) => {
+        return this.items[index].id;
+      });
+      this.$refs.deleteDialog.showDialog().then((confirmation) => {
+        if (confirmation === "ok") {
+          this.deleteItems(ids);
+        }
+      });
+    },
+    onClickDelete(item) {
+      this.$refs.deleteDialog.showDialog().then((confirmation) => {
+        if (confirmation === "ok") {
+          this.deleteItems([item.id]);
+        }
+      });
+    },
+
+    fetchData() {
+      // TODO: Loading
+      let query = `limit=${this.pageSize}`;
+      const offset = this.pageSize * (this.currentPage - 1);
+      query = query + `&offset=${offset}`;
+      this.$http
+        .get(`api/v1/admin/job-categories?${query}`)
+        .then((response) => {
+          const { data, meta } = response.data;
+          this.items = data;
+          this.total = meta.total;
+
+          if (this.total > this.pageSize) {
+            this.showPaginator = true;
+            this.pages = Math.floor(this.total / this.pageSize) + 1;
+          } else {
+            this.showPaginator = false;
+          }
+        })
+        .catch((error) => {
+          console.log(error);
+        });
+    },
+    deleteItems(items) {
+      // TODO: Loading
+      if (items instanceof Array) {
+        this.$http
+          .delete("api/v1/admin/job-categories", {
+            data: { ids: items },
+          })
+          .then(() => {
+            this.resetDataTable();
+            this.fetchData();
+          })
+          .catch((error) => {
+            console.log(error);
+          });
+      }
+    },
+    resetDataTable() {
+      this.$refs.dTable.checkedItems = [];
```

Comment:
I feel we are overusing refs. what happen to `this.checkedItems` state?

Reply:
i think it's not two way binded. changing the data table component's inside `checkedItems` state does change the local component's `checkedItems`, but changing it withing the local component didn't work. had to use a template ref. will check from oxd component's side.

Reply:
Vue is not two-way binding .. use compute{} and data{} wisely .. then component automatically update

Reply:
This has to be fixed from OXD side, i will check this.